jwst/tweakreg/tests/test_tweakreg.py

```diff
@@ -21,6 +29,33 @@ def dummy_source_catalog():
     return catalog
 
 
+@pytest.mark.parametrize("inplace", [True, False])
+def test_rename_catalog_columns(dummy_source_catalog, inplace):
+    renamed_catalog = tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+
+    # if testing inplace, check the input catalog
+    if inplace:
+        catalog = dummy_source_catalog
+    else:
+        catalog = renamed_catalog
+
+    assert 'xcentroid' not in catalog.colnames
+    assert 'ycentroid' not in catalog.colnames
+    assert 'x' in catalog.colnames
+    assert 'y' in catalog.colnames
+
+
+@pytest.mark.parametrize("missing", ["x", "y", "xcentroid", "ycentroid"])
+def test_rename_catalog_columns_invalid(dummy_source_catalog, missing):
+    # if the column we want to remove is not in the table, first run
+    # rename to rename columns this should add the column we want to remove
+    if missing not in dummy_source_catalog.colnames:
+        tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+    dummy_source_catalog.remove_column(missing)
+    with pytest.raises(ValueError, match="catalogs must contain"):
+        tweakreg_step._rename_catalog_columns(dummy_source_catalog)
+
+
 @pytest.mark.parametrize("offset, is_good", [(1 / 3600, True), (11 / 3600, False)])
 def test_is_wcs_correction_small(offset, is_good):
```

Comment:
Add doctsting with the purpose of the test here, too.

Reply:
See: https://github.com/spacetelescope/jwst/pull/8424#discussion_r1575026428